### PR TITLE
Social: Add UTM settings and toggles

### PR DESCRIPTION
**Pull request by gmjuhasz**
Fixes https://github.com/Automattic/jetpack-reach/issues/674

## Proposed changes:

* Adds a new setting and toggle for it to the Jetpack Social admin page and to Jetpack settings
* This setting will be used to toggle the UTM params on the sites

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to JP settings and to the Social Admin page
* Test that the toggle works as expected
* Also check the option on the local backend

CleanShot 2024-11-04 at 14 36 38 png

CleanShot 2024-11-04 at 14 36 20 png

### CODE REVIEW

**Comment by manzoorwanijk** [comment]
Instead of just a Boolean flag, should we make it an object like SIG settings with `enabled` being the only property for now? That will make it easier to add more related options in future.

Something like this

```json
"jetpack_social_utm_config": {
    "enabled": true
}
```

File: projects/plugins/jetpack/class.jetpack-gutenberg.php
Comment (manzoorwanijk) [inline projects/plugins/jetpack/class.jetpack-gutenberg.php]: We don't want to add this data to the old initial state, which is no longer used. All of our initial state is now at a single place - `projects/packages/publicize/src/class-publicize-script-data.php`

```suggestion
```

File: projects/plugins/social/src/class-jetpack-social.php
Comment (manzoorwanijk) [inline projects/plugins/social/src/class-jetpack-social.php]: Same here, this is not needed anymore.

```suggestion
```

**Comment by gmjuhasz** [comment]
> Instead of just a Boolean flag, should we make it an object like SIG settings with `enabled` being the only property for now? That will make it easier to add more related options in future.
> 
> Something like this
> 
> ```json
> "jetpack_social_utm_config": {
>     "enabled": true
> }
> ```

Done in 4f590a20d2dc158c382346930874bb63b2c788b0, thanks @manzoorwanijk

File: projects/plugins/jetpack/_inc/client/sharing/features/utm-toggle-section.jsx
Comment (manzoorwanijk) [inline projects/plugins/jetpack/_inc/client/sharing/features/utm-toggle-section.jsx]: A comma is better here

```suggestion
						"UTM parameters are tags added to links to help track where website visitors come from, improving our understanding of how content is shared. Don't worry, it doesn't change the experience or the link destination!",
```

**Review by manzoorwanijk — DISMISSED** [review]
This works well. I just have an optional suggestion.

File: projects/packages/publicize/src/jetpack-social-settings/class-settings.php
Comment (manzoorwanijk) [inline projects/packages/publicize/src/jetpack-social-settings/class-settings.php]: I think this is no longer needed because this `get_settings` method is used only for the initial state. Removing this will also fix the failing PHP unit tests.

File: projects/plugins/social/src/js/components/utm-toggle/index.tsx
Comment (manzoorwanijk) [inline projects/plugins/social/src/js/components/utm-toggle/index.tsx]: Need a comma here as well
```suggestion
					"UTM parameters are tags added to links to help track where website visitors come from, improving our understanding of how content is shared. Don't worry, it doesn't change the experience or the link destination!",
```

File: projects/plugins/social/src/js/components/utm-toggle/index.tsx
Comment (manzoorwanijk) [inline projects/plugins/social/src/js/components/utm-toggle/index.tsx]: Sorry, you were too fast to commit the previous suggestion 😄 

```suggestion
					"UTM parameters are tags added to links to help track where website visitors come from, improving our understanding of how content is shared. Don't worry, it doesn't change the experience or the link destination!",
```

Comment (gmjuhasz) [reply projects/plugins/social/src/js/components/utm-toggle/index.tsx]: Haha :D

File: projects/packages/publicize/src/jetpack-social-settings/class-settings.php
Comment (gmjuhasz) [reply projects/packages/publicize/src/jetpack-social-settings/class-settings.php]: I'll remove it then I thought we still need it

Comment (gmjuhasz) [reply projects/packages/publicize/src/jetpack-social-settings/class-settings.php]: I removed it, I'll give it another test if it works well but I'll finish the backend stuff first. If you get to test it before feel free to approve if looks good, thanks

**Review by manzoorwanijk — DISMISSED** [review]

File: projects/js-packages/publicize-components/src/social-store/types.ts
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/social-store/types.ts]:
```suggestion
	jetpack_social_utm_settings: UtmSettingsConfig;
```

**Review by manzoorwanijk — DISMISSED** [review]

**Review by manzoorwanijk — APPROVED** [review]

**Pull request merged**
